### PR TITLE
A32: Implement all ASIMD register comparison types

### PR DESCRIPTION
**Pull request by lioncash**
Implements:

- VCEQ (integer and FP)
- VCGE (integer and FP)
- VCGT (integer and FP)
- VACGE
- VACGT

register variants.

### CODE REVIEW

**Comment by lioncash** [comment]
Failure is due to a buildbot timeout

File: src/frontend/A32/translate/impl/asimd_three_same.cpp [hunk 1]
```diff
@@ -57,6 +65,79 @@ bool FloatingPointInstruction(ArmTranslatorVisitor& v, bool D, bool sz, size_t V
...
+    const auto d = ToVector(Q, Vd, D);
+    const auto m = ToVector(Q, Vm, M);
+    const auto n = ToVector(Q, Vn, N);
+
+    const auto reg_n = v.ir.GetVector(n);
+    const auto reg_m = v.ir.GetVector(m);
+    const auto result = [&] {
+        switch (comparison) {
+        case Comparison::GE:
+            return v.ir.FPVectorGreaterEqual(32, reg_n, reg_m);
```
Comment (MerryMage) [inline src/frontend/A32/translate/impl/asimd_three_same.cpp:124]:
```suggestion
            return v.ir.FPVectorGreaterEqual(32, reg_n, reg_m, false);
```

And the same for alll the other FP calls except for FPVectorAbs.

File: src/frontend/A32/translate/impl/asimd_three_same.cpp [hunk 2]
```diff
@@ -57,6 +65,79 @@ bool FloatingPointInstruction(ArmTranslatorVisitor& v, bool D, bool sz, size_t V
...
+        case Comparison::GT:
+            return v.ir.FPVectorGreater(32, reg_n, reg_m);
+        case Comparison::EQ:
+            return v.ir.FPVectorEqual(32, reg_n, reg_m);
+        case Comparison::AbsoluteGE:
+            return v.ir.FPVectorGreaterEqual(32, v.ir.FPVectorAbs(32, reg_n), v.ir.FPVectorAbs(32, reg_m));
```
Comment (MerryMage) [inline src/frontend/A32/translate/impl/asimd_three_same.cpp:130]:
```suggestion
            return v.ir.FPVectorGreaterEqual(32, v.ir.FPVectorAbs(32, reg_n), v.ir.FPVectorAbs(32, reg_m), false);
```

File: src/frontend/A32/translate/impl/asimd_three_same.cpp [hunk 3]
```diff
@@ -57,6 +65,79 @@ bool FloatingPointInstruction(ArmTranslatorVisitor& v, bool D, bool sz, size_t V
...
+        case Comparison::AbsoluteGT:
+            return v.ir.FPVectorGreater(32, v.ir.FPVectorAbs(32, reg_n), v.ir.FPVectorAbs(32, reg_m));
```
Comment (MerryMage) [inline src/frontend/A32/translate/impl/asimd_three_same.cpp:132]:
```suggestion
            return v.ir.FPVectorGreater(32, v.ir.FPVectorAbs(32, reg_n), v.ir.FPVectorAbs(32, reg_m), false);
```

File: src/frontend/A32/translate/impl/asimd_three_same.cpp (hunk 1 above)
Comment (lioncash) [reply src/frontend/A32/translate/impl/asimd_three_same.cpp]: Done

**Pull request merged**